app/frontend/src/components/Datepicker.tsx

```diff
@@ -3,13 +3,32 @@ import {
   KeyboardDatePicker,
   MuiPickersUtilsProvider,
 } from "@material-ui/pickers";
+import { CHANGE_DATE } from "features/constants";
 import { Control, Controller } from "react-hook-form";
 
+import { dateFormats } from "./constants";
+
+const getLocaleFormat = () => {
+  let locale =
+    navigator.language in dateFormats
+      ? dateFormats[navigator.language as keyof typeof dateFormats]
+      : "DD/MM/YYYY";
+  //convoluted way to convert and single letter date formats to double letter
+  //eg. "M/D/YYYY" to "MM/DD/YYYY"
+  //this is because keyboard typing is a total disaster for single letter formats
+  locale = locale
+    .replace("DD", "D")
+    .replace("MM", "M")
+    .replace("D", "DD")
+    .replace("M", "MM");
```

Comment:
```typescript
    .replace("D", "DD")
    .replace("M", "MM")
    .replace("DD", "D")
    .replace("MM", "M");
```
Would we not want to swap it around like this? So it goes 'DD' -> 'DDDD' -> 'DD' (same for 'M') for date formats that are already correct?

Reply:
Why? It's the same in the end, no?

Reply:
Oh yeah never mind, when I first read this I thought the ordering might mess up things. This still feels a bit weird and confusing... can we not just change the value in the mapping directly?